File: Estrutura_Repetição/Java/Ex1116/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();

        for (int i = 0; i < n; ++i) {
            int x = sc.nextInt();
            int y = sc.nextInt();

            if (y == 0) {
                System.out.println("divisao impossivel");
            } else if (x == 0) {
                System.out.println("0.0");
            } else {
                double valor = (double)x / y;
                System.out.printf("%.1f\n", valor);
            }
        }

        sc.close();
    }
}

File: Estrutura_Repetição/C/Ex1116/Main.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);

    for (int i = 0; i < n; ++i) {
        int x, y;
        scanf("%d %d", &x, &y);

        if (y == 0) {
            printf("divisao impossivel\n");
        } else if (x == 0) {
            printf("%.1f\n", 0.0);
        } else {
            double valor = (double)x / y;
            printf("%.1f\n", valor);
        }
    }

    return 0;
}